### ISSUE TITLE
Cannot remove subscription metadata

### ISSUE THREAD

**Issue by omgitsotis** [stripe/stripe-go#511]
I'm trying to remove some metadata on a subscription, and before I simply provided an empty string, but now that does not seem to do anything. 

My Code
```go
params := &stripe.SubParams{}
params.AddMeta("Test", "")
sub, _ := sc.Subs.Update("sub_Bz2wy74jBPoaUD", params)
log.Debugf("METADATA VALUES %v", sub.Meta)
```
I have looked at the request that is created and it does not add the metadata field if the value is empty. I cannot find anything to suggest that removing metadata has changed, so if it has could you kindly show me where that is.

**Comment by ob-stripe** [comment]
Hey @omgitsotis, thanks for the report.

I think this issue is caused by #446. The [string encoder function](https://github.com/stripe/stripe-go/blob/e9fd39d55f82a644ba52d27705c820bdf8da7585/form/form.go#L322) will only encode empty strings if it receives a [string pointer](https://github.com/stripe/stripe-go/blob/e9fd39d55f82a644ba52d27705c820bdf8da7585/form/form.go#L343). Since `Meta` is declared as `map[string][string]`, it's not possible to pass a pointer and thus not possible to encode empty values.

Assigning to @brandur-stripe to confirm and hopefully come up with a fix!

**Comment by brandur-stripe** [comment]
@ob-stripe Yep, that sounds accurate — thanks for the diagnosis! I added a fix in #512.

**Comment by brandur-stripe** [comment]
Released as 28.10.1.
